modules/quadkey/redshift/sql/QUADINT_FROMGEOGPOINT.sql

```diff
@@ -8,5 +8,5 @@ CREATE OR REPLACE FUNCTION @@RS_PREFIX@@carto.QUADINT_FROMGEOGPOINT
 RETURNS BIGINT
 STABLE
 AS $$
-    SELECT @@RS_PREFIX@@carto.QUADINT_FROMLONGLAT(ST_X($1), ST_Y($1), $2)
+    SELECT @@RS_PREFIX@@carto.QUADINT_FROMLONGLAT(ST_X(ST_SetSRID($1, 4326)), ST_Y(ST_SetSRID($1, 4326)), $2)
```

Comment:
But ST_SetSRID blindly sets the SRID. If the geometry is originally in other reference we're getting the wrong coordinates. We should use ST_TRANSFORM to compute the correct 4326 coordinates. But it fails if the geometry doesn't have a SRID (0). What I ended up doing in PG is:
`CASE WHEN ST_SRID($1) = 0 THEN ST_SetSRID($1, 4326) ELSE ST_Transform($1, 4326) END`

Reply:
Updated